File: backend/app/routes/courses.py
from fastapi import APIRouter
from fastapi import Depends

from sqlalchemy.orm import Session

from app.database import get_db
from app.schemas.course import CourseCreate
from app.services.course_service import (
    create_course,
    get_courses,
)


router = APIRouter(
    prefix="/courses",
    tags=["Courses"]
)


@router.get("/")
def list_courses(
    db: Session = Depends(get_db)
):

    return get_courses(db)


@router.post("/")
def create_new_course(
    course: CourseCreate,
    db: Session = Depends(get_db)
):

    return create_course(db, course)

File: frontend/src/api/courses.js
import api from "./client";

export const getCourses = () =>
    api.get("/courses");

export const getCourse = (id) =>
    api.get(`/courses/${id}`);

export const createCourse = (course) =>
    api.post("/courses", course);

export const updateCourse = (id, course) =>
    api.post(`/courses/${id}`, course);

export const deleteCourse = (id) =>
    api.delete(`/courses/${id}`);